main.py

```diff
@@ -246,16 +246,16 @@ def evaluate_and_write(args, model, tasks, splits_to_write):
         evaluate.write_preds(tasks, val_preds, args.run_dir, 'val',
                              strict_glue_format=args.write_strict_glue_format)
     if 'test' in splits_to_write:
-        _, te_preds = evaluate.evaluate(
-            model, tasks, args.batch_size, args.cuda, "test")
+        te_results, te_preds = evaluate.evaluate(model, tasks, args.batch_size, args.cuda, "test")
         evaluate.write_preds(tasks, te_preds, args.run_dir, 'test',
                              strict_glue_format=args.write_strict_glue_format)
     run_name = args.get("run_name", os.path.basename(args.run_dir))
 
     results_tsv = os.path.join(args.exp_dir, "results.tsv")
     log.info("Writing results for split 'val' to %s", results_tsv)
     evaluate.write_results(val_results, results_tsv, run_name=run_name)
-
+    if 'test' in splits_to_write:
+        evaluate.write_results(te_results, results_tsv, run_name=run_name)
```

Comment:
Why do we only write *results* if splits_to_write is set?

Reply:
Not sure if I understand, predictions are written in `evaluate.write_preds(task, te_preds, ...)` above.